src/test/operators/jit_operator/operators/jit_aggregate_test.cpp

```diff
@@ -341,4 +341,40 @@ TEST_F(JitAggregateTest, EmptyInputTableNoGroupbyColumns) {
   EXPECT_TABLE_EQ_ORDERED(output_table, expected_output_table);
 }
 
+TEST_F(JitAggregateTest, UpdateNullableInformationBeforeSpecialization) {
+  // The nullable information of the aggregate and group by expressions must be updated before specialization
+
+  const auto input_table = Table::create_dummy_table(TableColumnDefinitions{{"a", DataType::Int, false},
+                                                                            {"b", DataType::Long, true},
+                                                                            {"c", DataType::Float, false},
+                                                                            {"d", DataType::String, true}});
+
+  // Create tuple entries without setting the correct nullable information
+  const JitTupleEntry tuple_entry_a{DataType::Int, true, 0};
+  const JitTupleEntry tuple_entry_b{DataType::Long, true, 1};
+  const JitTupleEntry tuple_entry_c{DataType::Float, true, 2};
+  const JitTupleEntry tuple_entry_d{DataType::String, true, 3};
+
+  JitAggregate jit_aggregate;
+  jit_aggregate.add_aggregate_column("min_a", tuple_entry_a, AggregateFunction::Min);
+  jit_aggregate.add_aggregate_column("count_b", tuple_entry_b, AggregateFunction::Count);
+  jit_aggregate.add_groupby_column("c", tuple_entry_c);
+  jit_aggregate.add_groupby_column("d", tuple_entry_d);
+
+  // Update nullable information
+  // a and c are not nullable, b and d are nullable
+  std::vector<bool> tuple_nullable_information{false, true, false, true};
+  jit_aggregate.before_specialization(*input_table, tuple_nullable_information);
+
+  const auto& aggregate_columns = jit_aggregate.aggregate_columns();
+  EXPECT_FALSE(aggregate_columns[0].tuple_entry.is_nullable);
```

Comment:
This looks incorrect, see above

Reply:
Add it to the test and explain which step (before or after the aggregate) you are testing?

Reply:
I added this to the test and added an explaining comment.